nvflare/app_common/np/recipes/lr/fedavg.py

```diff
@@ -0,0 +1,137 @@
+            num_clients: Optional[int] = None,
+            min_clients: int = 0,
+            num_rounds: int = 2,
+            damping_factor=0.8,
+            train_script: str,
+            train_args: str = "",
+            launch_external_process=False,
+            command: str = "python3 -u",
+    ):
+        # Validate inputs internally
+        v = _FedAvgValidator(
+            name=name,
+            initial_model=initial_model,
+            clients=clients,
+            num_clients=num_clients,
+            min_clients=min_clients,
+            num_rounds=num_rounds,
+            damping_factor=damping_factor,
+            train_script=train_script,
+            train_args=train_args,
+            launch_external_process=launch_external_process,
+            command=command
+        )
+
+        self.name = v.name
+        self.initial_model = v.initial_model
+        self.clients = v.clients
+        self.num_clients = v.num_clients
+        self.min_clients = v.min_clients
+        self.num_rounds = v.num_rounds
+        self.damping_factor = v.damping_factor
+        self.initial_model = v.initial_model
+        self.clients = v.clients
+        self.train_script = v.train_script
+        self.train_args = v.train_args
+        self.launch_external_process = v.launch_external_process
+        self.command = v.command
+
+        # Create FedJob.
+        job = FedJob(name=self.name)
+        persistor_id = job.to_server(LRModelPersistor(n_features=13), id="lr_persistor")
+
+        # Send custom controller to server
+        controller = FedAvgLR(
+            n_clients=0,  # for all clients
+            min_clients=self.min_clients,
+            damping_factor=self.damping_factor,
+            num_rounds=self.num_rounds,
+            initial_model=self.initial_model,
+            persistor_id=persistor_id,
+        )
+        job.to(controller, "server")
+
+        # Send TBAnalyticsReceiver to server for tensorboard streaming.
+        analytics_receiver = TBAnalyticsReceiver()
+        job.to_server(
+            id="receiver",
+            obj=analytics_receiver,
+        )
+        convert_to_fed_event = ConvertToFedEvent(events_to_convert=[ANALYTIC_EVENT_TYPE])
+
+        # Add clients
+        if self.clients is None:
+            clients = [f"site-{i + 1}" for i in range(self.num_clients)]
+        else:
+            clients = self.clients
+
+        for client in clients:
+            job.to(id="event_to_fed", obj=convert_to_fed_event, target=client)
+            runner = ScriptRunner(
+                script=self.train_script,
+                script_args=self.train_args,
+                launch_external_process=self.launch_external_process,
+                command=self.command,
+                framework=FrameworkType.RAW,
+                server_expected_format=ExchangeFormat.RAW,
+                params_transfer_type=TransferType.FULL
+            )
+            job.to(runner, client)
```

Comment:
I think "job.to_clients" should work? if it's not working we can merge this and I can check later as well